bounded_counter/count_min_sketch.py

```diff
@@ -1,9 +1,10 @@
 import os
 import numpy
+import xxhash
 
 
 def hash_key(i, key):
-    return hash(str(i) + str(key))
+    return xxhash.xxh64(str(key), seed=i).intdigest()
```

Comment:
Better use one of the standard hash libraries that I linked to in the beginning. I never heard of `xxhash` (doesn't mean it's bad, just maturity/support unknown).

But can you explain why we use custom hashing at all? And why the previous code has the intermediate step of converting to `str`? What's wrong with the built-in `hash`?

Function needs a docstring (what is `i`? not a good variable name)

Reply:
Which links to hash libraries are you referring to?
I was going to use [pyhash](https://github.com/flier/pyfasthash), but I had trouble installing it on both Windows and Unix. 
We can use any good hash function as long as it is deterministic and cheaply seedable.

Python's hash is deliberately initialized with a random seed so that dictionary attacks are impossible. If we used that, we couldn't pickle and reuse stored dictionaries.
Furthermore, we want some fast library implemented in C to skip python altogether.

Will document better.